packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -50,6 +50,20 @@ export abstract class AbstractResourcePreferenceProvider implements PreferencePr
         return this.preferences;
     }
 
+    async setPreference(key: string, value: string): Promise<void> {
+        const resource = await this.resource;
+        if (resource.saveContents) {
+            let preferences = this.preferences;
+            if (preferences) {
+                preferences[key] = value;
+            } else {
+                preferences = {[key]: value};
+            }
+            resource.saveContents(JSON.stringify(preferences, undefined, 3));
```

Comment:
You should use an indentation from the original file, e.g. with https://www.npmjs.com/package/detect-indent

Reply:
done